vector/src/main/java/im/vector/adapters/VectorUnknownDevicesAdapter.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright 2017 Vector Creations Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.adapters;
+
+import android.content.Context;
+
+import org.matrix.androidsdk.util.Log;
+
+import android.util.Pair;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseExpandableListAdapter;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.matrix.androidsdk.crypto.data.MXDeviceInfo;
+
+import java.util.List;
+
+import im.vector.R;
+
+/**
+ * This class displays a list of members to create a room.
+ */
+public class VectorUnknownDevicesAdapter extends BaseExpandableListAdapter {
+
+    private static final String LOG_TAG = "VRoomSummaryAdapter";
+
+    // devices verification listener
+    public interface IVerificationAdapterListener {
+        /**
+         * Verify device button handler
+         *
+         * @param aDeviceInfo device info
+         */
+        void OnVerifyDeviceClick(MXDeviceInfo aDeviceInfo);
+
+        /**
+         * Block device button handler
+         *
+         * @param aDeviceInfo device info
+         */
+        void OnBlockDeviceClick(MXDeviceInfo aDeviceInfo);
+    }
+
+    // context
+    private Context mContext;
+    // layout inflater
+    private LayoutInflater mLayoutInflater;
+    // devices list
+    private List<Pair<String, List<MXDeviceInfo>>> mUnknownDevicesList;
+    // listener
+    private IVerificationAdapterListener mListener;
+
+    /**
+     * Constructor
+     *
+     * @param aContext       the context
+     * @param unknownDevices the unknown devices list
+     */
+    public VectorUnknownDevicesAdapter(Context aContext, List<Pair<String, List<MXDeviceInfo>>> unknownDevices) {
+        // init internal fields
+        mContext = aContext;
+        mLayoutInflater = LayoutInflater.from(mContext);
+        mUnknownDevicesList = unknownDevices;
```

Comment:
Initialize `mUnknownDevicesList` to empty array in case `unknownDevices` is null?